### PR TITLE
feat: [#441] Add Fluent Json Api to test Json response [#4]

### PR DESCRIPTION
**Pull request by kkumar-gcc**
## 📑 Description

RelatedTo https://github.com/goravel/goravel/issues/441

## Summary by CodeRabbit

- **New Features**
	- Enhanced JSON assertion capabilities with new methods for checking key existence, value matching, and absence of keys.
	- Introduced a fluent interface for more complex JSON assertions.
	- Added methods for improved handling of HTTP requests, including token management and basic authentication.

- **Bug Fixes**
	- Improved error handling and assertion checks in response handling.

- **Tests**
	- Added comprehensive test cases for various HTTP response statuses and JSON assertions, increasing test coverage.
	- Introduced unit tests for validating JSON structures and assertion methods.

## ✅ Checks

- [ ] Added test cases for my code

### CODE REVIEW

File: testing/assertable_json.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	contractstesting "github.com/goravel/framework/contracts/testing"
+	"github.com/goravel/framework/support/maps"
+)
+
+type AssertableJsonImpl struct {
```
Comment (hwbrzzl) [inline testing/assertable_json.go:14]: It's different packages with `AssertableJson interface`, so the struct name can be `AssertableJson` directly.

```suggestion
type AssertableJson struct {
```

File: testing/assertable_json.go
Comment (hwbrzzl) [inline testing/assertable_json.go:1]: Add unit tests for this file.

**Review by hwbrzzl — COMMENTED** [review]
Pretty good 👍

File: testing/test_request.go [hunk 2]
```diff
@@ -59,6 +66,23 @@ func (r *TestRequest) WithContext(ctx context.Context) contractstesting.TestRequ
...
 
+func (r *TestRequest) WithToken(token string, ttype ...string) contractstesting.TestRequest {
+	tt := "Bearer"
+	if len(ttype) > 0 {
+		tt = ttype[0]
+	}
+	return r.WithHeader("Authorization", fmt.Sprintf("%s %s", tt, token))
+}
+
+func (r *TestRequest) WithBasicAuth(username, password string) contractstesting.TestRequest {
```
[the remark this answers is not in the document]
Comment (kkumar-gcc) [reply testing/test_request.go:77]: Usage of this be added in next PR. Some time api expect authorization. Uset can use this method there.

File: contracts/testing/assertable_json.go [hunk 3]
```diff
@@ -2,4 +2,15 @@ package testing
...
+	Has(key string) AssertableJSON
+	HasAll(keys []string) AssertableJSON
+	HasAny(keys []string) AssertableJSON
+	Missing(key string) AssertableJSON
+	MissingAll(keys []string) AssertableJSON
+	Where(key string, value any) AssertableJSON
+	WhereNot(key string, value any) AssertableJSON
+	Each(key string, callback func(AssertableJSON)) AssertableJSON
+	First(key string, callback func(AssertableJSON)) AssertableJSON
+	HasWithScope(key string, length int, callback func(AssertableJSON)) AssertableJSON
```
[the remark this answers is not in the document]
Comment (kkumar-gcc) [reply contracts/testing/assertable_json.go:15]: I have already added. Plz find in `assertable_json_test`

**Review by hwbrzzl — APPROVED** [review]
Great

**Pull request merged**